eeps/eep-0057.md

```diff
@@ -0,0 +1,227 @@
+-----------
+
+When the supervisor notices the crash of a child (the offender)...
+
+1. it terminates all _running_ siblings and marks them and as well as the offending
+   child as restarting
+2. it starts a timer to send a message to itself after the _maximum_ of the delays
+   of the offending child and the siblings terminated in (1)
+3. it resumes its receive loop
+4. when the timed message arrives, it will trigger the actual restarts of the
+   offending child and the siblings terminated in (1), in proper order
+
+If any child fails to start, this procedure is repeated, with the failed
+child becoming the new offender.
+
+By using the maximum of the delays in (2), the children are restarted at the same time,
+as a unit, without intermittend delays.
+
+While the delay between a crash and the first restart attempt is constant, the delays
+_between_ restart _retries_ are not necessarily constant. The children behind the failing one
+were not tried, and by having used the maximum delay in (2), their delays need not be
+considered when retrying. Put simply, only the delays of the children before the failing
+one need to be considered when retrying a restart.
+
+
+
+rest_for_one
+------------
+
+When the supervisor notices the crash of a child (the offender)...
+
+1. it terminates all _running_ siblings behind the offending child and marks
+   them as well as the offending child as restarting
+2. it starts a timer to send a message to itself after the _maximum_ of the delays
+   of the offending child and the siblings terminated in (1)
+3. it resumes its receive loop
+4. when the timed message arrives, it will trigger the actual restarts of the
+   offending child and the siblings terminated in (1), in proper order
+
+If any child fails to start, this procedure is repeated, with the failed
+child becoming the new offender.
+
+While the maximum of the delays is also used in (2) in this strategy, the
+guarantees it provides, concerning restarting at the same time as a unit
+without intermittend delays, are weaker: When one of the children fails to
+restart, the restarting will resume from there, but after the respective delay.
+
+Like in the one_for_all strategy, the delay between a crash and the first
+restart attempt is constant, but the delays are not necessarily constant. In
+rest_for_one, we can even say that the delays between retries are either the
+same or decreasing. The children before the offending child are running and so
+need not be considered. The children behind the offending child have not been
+tried, and having used the maximum delay in (2), they need not be considered also.
+Put simply, only the delay of the offending child needs to be considered when
+retrying a restart.
+
+A special case in rest_for_one is that one of the still running children may
+crash while children further back are in delay. In this case, the calculation of
+the new delay should take the remainder of the already running delay into consideration.
```

Comment:
Suggestion: make it explicit that the delayed children and the newly failed children will be grouped.

```suggestion
crash while children further back are in delay. In this case, the new failed children
will be grouped with the currently delayed children, and the calculation of the new
delay should take the remainder of the already running delay into consideration.
```

Reply:
Thanks :)